marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/Step.java

```diff
@@ -133,15 +133,19 @@ public static Step deserialize(JsonNode json) {
         Step step = new Step();
 
         JSONObject jsonObject = new JSONObject(json);
-        step.setName(jsonObject.getString("name"));
+        step.setStepDefinitionName(jsonObject.getString("stepDefinitionName"));
+        step.setStepDefinitionType(StepDefinition.StepDefinitionType.getStepDefinitionType(jsonObject.getString("stepDefinitionType")));
+        String stepName = jsonObject.getString("name");
+        if (stepName == null) {
+            stepName = step.getStepDefinitionName();
```

Comment:
Is step name optional?

Reply:
Yes, but that's not encouraged.

PKey for the step is the #.

Reply:
Good question. The backend doesn't require it and I don't believe the middle-tier does either, but Quick Start seems to.

Reply:
We seem to track both step name and step-def name in prov.
Should we make it required as going forward when we have a step-def bank the step name will come in handy to track the relation between both?

Reply:
Yeah. Going forward we should.